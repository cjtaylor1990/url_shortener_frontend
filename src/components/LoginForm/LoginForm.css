
input {
  background-color: lightgray;
  display: block;
  margin-bottom: 1em;
}

button.login {
  background-color: lightskyblue;
  border-radius: 0.25rem;
  display:block;
}