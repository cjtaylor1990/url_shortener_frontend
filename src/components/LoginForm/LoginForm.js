import React from 'react';
import './LoginForm.css';

class LoginForm extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			username: undefined,
			password: undefined
		};

		this.handleUsernameChange = this.handleUsernameChange.bind(this);
		this.handlePasswordChange = this.handlePasswordChange.bind(this);
		this.handleLogin = this.handleLogin.bind(this);
	};

	handleUsernameChange(event) {
		this.setState({username: event.target.value});
	};

	handlePasswordChange(event) {
		this.setState({password: event.target.value});
	};

	handleLogin(event) {
		this.props.authenticate(this.state.username, this.state.password);
		event.preventDefault();
	};

	render() {
    	return(
			<div class="loginForm">
				<h1> URL Shortener </h1>
				<input type="text" class="username" placeholder="Username" onChange={this.handleUsernameChange}/>
				<input type="password" class="password" placeholder="Password" onChange={this.handlePasswordChange}/>
				<button type="submit" class="login" onClick={this.handleLogin}> Log In </button>
			</div>
		);
	};

};

export default LoginForm;
