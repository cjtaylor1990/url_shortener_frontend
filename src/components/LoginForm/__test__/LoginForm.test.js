import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import LoginForm from '../LoginForm.js';

afterEach(cleanup);

describe('LoginForm initially', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
            
        ReactDOM.render(<LoginForm />, div);
            
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(<LoginForm/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has blank input for username and password', () => {
        const { getByTestId } = render(<LoginForm />);

        const usernameInput = getByTestId('loginFormUsername');
        const passwordInput = getByTestId('loginFormPassword');

        expect(usernameInput).toHaveTextContent('');
        expect(passwordInput).toHaveTextContent('');

    });
    
});