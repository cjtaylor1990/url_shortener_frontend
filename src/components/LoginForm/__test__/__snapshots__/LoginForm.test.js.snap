// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm initially matches snapshot 1`] = `
<div
  className="loginForm"
  data-testid="loginForm"
>
  <input
    className="username"
    data-testid="loginFormUsername"
    onChange={[Function]}
    placeholder="Username"
    type="text"
  />
  <input
    className="password"
    data-testid="loginFormPassword"
    onChange={[Function]}
    placeholder="Password"
    type="password"
  />
  <button
    className="login"
    data-testid="loginFormLogin"
    onClick={[Function]}
    type="submit"
  >
     Log In 
  </button>
</div>
`;
